### PR TITLE
Refactor rgblight_reconfig.h

### PR DESCRIPTION
**Pull request by mtei**
## Description

In #3582, rgblight_reconfig.h had to be newly created. ~~Now, the build system of qmk_firmware has a post_cofig feature, so that what was done in rgblight_reconfig.h can now be realized in rgblight_post_config.h.~~ Now, merged the contents of rgblight_reconfig.h into rgblight.h to make the source files under `tmk_core/` more concise.

**This commit does not change the build result.**

move contents of rgblight_reconfig.h to rgblight.h.

The following changes were made to rgblight.h.

```diff
+#ifdef RGBLIGHT_USE_TIMER
 void rgblight_task(void);

 void rgblight_timer_init(void);
 void rgblight_timer_enable(void);
 void rgblight_timer_disable(void);
 void rgblight_timer_toggle(void);
+#else
+#define rgblight_task()
+#define rgblight_timer_init()
+#define rgblight_timer_enable()
+#define rgblight_timer_disable()
+#define rgblight_timer_toggle()
+#endif
```

The following changes were made to tmk_core/common/avr/suspend.c, tmk_core/common/chibios/suspend.c, tmk_core/protocol/chibios/main.c, tmk_core/protocol/lufa/lufa.c, tmk_core/protocol/vusb/main.c.

```diff
-#    ifdef RGBLIGHT_ANIMATIONS
     rgblight_timer_enable();
-#    endif
```
```diff
-#if defined(RGBLIGHT_ANIMATIONS) && defined(RGBLIGHT_ENABLE)
+#if defined(RGBLIGHT_ENABLE)
         rgblight_task();
 #endif
```

Testing script
```shell
  # build on master
  git checkout `git merge-base master refactor_rgblight_reconfig.h`
  git log -1 --oneline --decorate > /tmp/master_md5.txt

  # RGBLIGHT_ENABLE = no
  make HELIX=verbose helix/rev2:default:clean
  make HELIX=verbose helix/rev2:default
  md5 helix_rev2_default.hex >> /tmp/master_md5.txt

  # RGBLIGHT_ENABLE = yes, with animations
  make HELIX=verbose helix/rev2/back:default:clean
  make HELIX=verbose helix/rev2/back:default
  md5 helix_rev2_back_default.hex >> /tmp/master_md5.txt

  # RGBLIGHT_ENABLE = yes, without animations
  make HELIX=verbose,no_ani helix/rev2/back:default:clean
  make HELIX=verbose,no_ani helix/rev2/back:default
  md5 helix_rev2_back_default.hex >> /tmp/master_md5.txt

  # build on refactor_rgblight_reconfig.h
  git checkout refactor_rgblight_reconfig.h
  git log -4 --oneline --decorate > /tmp/branch_md5.txt

  # RGBLIGHT_ENABLE = no
  make HELIX=verbose helix/rev2:default:clean
  make HELIX=verbose helix/rev2:default
  md5 helix_rev2_default.hex >> /tmp/branch_md5.txt

  # RGBLIGHT_ENABLE = yes, with animations
  make HELIX=verbose helix/rev2/back:default:clean
  make HELIX=verbose helix/rev2/back:default
  md5 helix_rev2_back_default.hex >> /tmp/branch_md5.txt

  # RGBLIGHT_ENABLE = yes, without animations
  make HELIX=verbose,no_ani helix/rev2/back:default:clean
  make HELIX=verbose,no_ani helix/rev2/back:default
  md5 helix_rev2_back_default.hex >> /tmp/branch_md5.txt

  diff -u /tmp/master_md5.txt /tmp/branch_md5.txt
```

Test result:
```
--- /tmp/master_md5.txt 2020-01-04 15:10:24.000000000 +0900
+++ /tmp/branch_md5.txt 2020-01-04 15:10:45.000000000 +0900
@@ -1,4 +1,7 @@
-dd0407909 (HEAD) Fixed a small typo (#7743)
+e7105463e (HEAD -> refactor_rgblight_reconfig.h, origin/refactor_rgblight_reconfig.h) Edit the expression again
+88c13e151 Expressions that are too long are difficult to read, so wrap them.
+a270119df Moved contents of rgblight_reconfig.h to rgblight_post_config.h.
+dd0407909 Fixed a small typo (#7743)
 MD5 (helix_rev2_default.hex) = f360032edd522448366d471d8f4f8181
 MD5 (helix_rev2_back_default.hex) = 0c663acc6cccc44476b3b969ad22a48f
 MD5 (helix_rev2_back_default.hex) = e66b1195ff6d38e6e22c975b8ae42fd3
```
```
--- /tmp/master_md5.txt 2020-01-05 16:58:49.000000000 +0900
+++ /tmp/branch_md5.txt 2020-01-05 16:59:11.000000000 +0900
@@ -1,4 +1,8 @@
-dd0407909 (HEAD) Fixed a small typo (#7743)
+7df3b730f (HEAD -> refactor_rgblight_reconfig.h) remove `defined(RGBLIGHT_ANIMATIONS)` in `tmk_core/common/*/suspend.c`, `tmk_core/protocol/*/main.c`
+e7105463e (origin/refactor_rgblight_reconfig.h) Edit the expression again
+88c13e151 Expressions that are too long are difficult to read, so wrap them.
+a270119df Moved contents of rgblight_reconfig.h to rgblight_post_config.h.
+dd0407909 Fixed a small typo (#7743)
 MD5 (helix_rev2_default.hex) = f360032edd522448366d471d8f4f8181
 MD5 (helix_rev2_back_default.hex) = 0c663acc6cccc44476b3b969ad22a48f
 MD5 (helix_rev2_back_default.hex) = e66b1195ff6d38e6e22c975b8ae42fd3
```
```
--- /tmp/master_md5.txt 2020-01-21 05:55:03.000000000 +0900
+++ /tmp/branch_md5.txt 2020-01-21 05:55:24.000000000 +0900
@@ -1,4 +1,8 @@
-61dbb9267 (HEAD, qmk/master, origin/master, origin/HEAD, qmk_master, master) Invert Num Lock LED state (#7945)
+5a4588027 (HEAD -> refactor_rgblight_reconfig.h) remove 'defined(RGBLIGHT_ANIMATIONS)' in tmk_core/common/keyboard.c
+693bd87eb remove `defined(RGBLIGHT_ANIMATIONS)` in `tmk_core/common/*/suspend.c`, `tmk_core/protocol/*/main.c`
+757af486e Edit the expression again
+1d6b01f7b Expressions that are too long are difficult to read, so wrap them.
+e617fd6f3 Moved contents of rgblight_reconfig.h to rgblight_post_config.h.
 MD5 (helix_rev2_default.hex) = f360032edd522448366d471d8f4f8181
 MD5 (helix_rev2_back_default.hex) = e601bff0a51fdf18e0c105be6ba54868
 MD5 (helix_rev2_back_default.hex) = e66b1195ff6d38e6e22c975b8ae42fd3
```

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: quantum/rgblight_post_config.h [hunk 1]
```diff
@@ -3,3 +3,37 @@
...
+// for backward compatibility
+#    define RGBLIGHT_EFFECT_BREATHING
+#    define RGBLIGHT_EFFECT_RAINBOW_MOOD
+#    define RGBLIGHT_EFFECT_RAINBOW_SWIRL
+#    define RGBLIGHT_EFFECT_SNAKE
+#    define RGBLIGHT_EFFECT_KNIGHT
+#    define RGBLIGHT_EFFECT_CHRISTMAS
+#    define RGBLIGHT_EFFECT_STATIC_GRADIENT
+#    define RGBLIGHT_EFFECT_RGB_TEST
+#    define RGBLIGHT_EFFECT_ALTERNATING
```
Comment (drashna) [inline quantum/rgblight_post_config.h]: Could these two be moved to a different define, something like:

```
#ifdef RGBLIGHT_ANIMATIONS_TEST
#    define RGBLIGHT_EFFECT_RGB_TEST
#    define RGBLIGHT_EFFECT_ALTERNATING
#endif

Comment (mtei) [reply quantum/rgblight_post_config.h]: I found #7680 and #7669. However, I want to focus on refactoring rgblight_reconfig.h in this PR.

Comment (drashna) [reply quantum/rgblight_post_config.h]: Okay

**Comment by drashna** [comment]
Looks like there are some merge conflicts here, now.

**Comment by JohSchneider** [comment]
hm, travis fails on keyboard/mxss:default because of a missing include rgblight_reconfig.h :-S

**Pull request merged**